File: PolishPizza/delivery/src/main/java/pizzeria/deliveries/supplier/model/Supplier.java
package pizzeria.deliveries.supplier.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "suppliers")
@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class Supplier {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(unique = true, nullable = false)
    private UUID userProfileId;
    private String firstName;
    private String lastName;
    private String phoneNumber;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private SupplierStatus status = SupplierStatus.OFFLINE;
}

File: PolishPizza/admin/src/types/supplier.ts
import type { SupplierStatus } from "./enums";

export interface SupplierResponse {
    id: string;
    userProfileId: string;
    firstName: string;
    lastName: string;
    phoneNumber: string;
    status: SupplierStatus;
}

export interface SupplierChangeStatusRequest {
    status: SupplierStatus;
}

